### DISCUSSION TITLE
pathRewrite not matching regex pattern with variables

### DISCUSSION THREAD

**Discussion by alex996** [chimurai/http-proxy-middleware#538] [Ideas]
I am having an issue getting `pathRewrite` to match a regex pattern in `context` to variables in `target`.

### Expected behavior
It should map a regex pattern with expressions (e.g. `(.*)`) to variables (e.g. `$1`).

### Actual behavior
Proxy is not being created and the Express route yields 404.

### Setup

* http-proxy-middleware: ^0.18.0
* server: express@^4.16.3

#### proxy middleware configuration
```javascript
proxy('/user/(.*)/posts/(.*)', {
  pathRewrite: {
    '^/user/(.*)/posts/(.*)': '/api/user/$1/posts/$2'
  }
})
```

Is this functionality supported?

**Comment by chimurai** [comment]
Capture groups are not supported.

You can provide a custom function to do the path rewriting the way you want:
https://github.com/chimurai/http-proxy-middleware/blob/master/recipes/pathRewrite.md#custom-rewrite-function

**Comment by alex996** [comment]
@chimurai Thanks! Then never mind, I'll leave this as a feature request.

**Comment by chimurai** [comment]
Will consider adding it if more users have to need for this.

**Comment by gonzadocarmo** [comment]
+1

**Comment by mheatleyTMN** [comment]
+1

**Comment by rupanpaul** [comment]
+1

**Comment by howard-brian** [comment]
+1. This would be awesome

**Comment by jtaalmasco** [comment]
+1

**Comment by CrafterBoy** [comment]
+1

**Comment by johnnorro** [comment]
+1

**Comment by PodgeHn** [comment]
+1

**Comment by ColinFay** [comment]
+1, that would be dope

**Comment by tinyzimmer** [comment]
+1

**Comment by aserrabl** [comment]
+1

**Comment by cdll** [comment]
+1

**Comment by iain-b** [comment]
+1

**Comment by dillonlee8** [comment]
+1

**Comment by jorgesanux** [comment]
+1
